drivers: ssd1673: add DT properties for config registers, add SSD1608 support

jfischer-no opened the pull request:
The values for the registers like GDV, SDV and Border Waveform
depend on the panel and display controller. Add DT properties
and obtain such values from DT.

Use compatible property to identify display and select
proper LUT for display controller.

Add support for SSD1608 controller and GDE029A1 display.

jfischer-no committed:
drivers: ssd1673: add DT properties for configuration registers

The values for the registers like GDV, SDV and Boarder Waveform
depend on the panel and display controller. Add DT properties
and obtain such values from DT.

Signed-off-by: Johann Fischer <j.fischer@phytec.de>

jfischer-no committed:
drivers: ssd1673: remove unnecessary defines

Remove unnecessary defines.

Signed-off-by: Johann Fischer <j.fischer@phytec.de>

jfischer-no committed:
drivers: ssd1673: use compatible property to identify display

Use compatible property to identify display and select
proper LUT for display controller.

Signed-off-by: Johann Fischer <j.fischer@phytec.de>

jfischer-no committed:
drivers: ssd1673: add support for SSD1608 and GDE029A1

Add support for SSD1608 controller and GDE029A1 display.

Signed-off-by: Johann Fischer <j.fischer@phytec.de>

vanwinkeljan commented on drivers/display/ssd1673.c:
```diff
@@ -467,20 +467,24 @@ static int ssd1673_controller_init(struct device *dev)
-	tmp[1] = SSD1673_VAL_GDV_CTRL_B;
+	tmp[0] = DT_SOLOMON_SSD1673FB_0_CONTROLLER_GDV_A;
+#if defined(DT_SOLOMON_SSD1673FB_0_CONTROLLER_GDV_B)
+	tmp[1] = DT_SOLOMON_SSD1673FB_0_CONTROLLER_GDV_A;
```
Should this not be GDV_B instead of GDV_A?

vanwinkeljan commented:
@jfischer-phytec-iot definitely a good idea to put this info in DT.
Except the one small remark above I would like see the input from @galak  and @MaureenHelm on this topic as there is a proposal from @galak in PR https://github.com/zephyrproject-rtos/zephyr/pull/12457#issuecomment-457016502 on following the linux kernel way of handling display pannels.

aurel32 commented on drivers/display/ssd1673.c:
```diff
@@ -45,23 +45,35 @@ struct ssd1673_data {
+	0x00, 0x00, 0x00, 0x00, 0x13, 0x14, 0x44, 0x12,
+	0x00, 0x00, 0x00, 0x00, 0x00, 0x00
+};
+#endif
```
Maybe add a #else and #error in case no LUT is defined for the current device.

aurel32 committed:
drivers: ssd1673: add support for HINK E0154A05

Add support for HINK E0154A05 display, used notably in the Waveshare
1.54inch e-Paper Module.

Signed-off-by: Aurelien Jarno <aurelien@aurel32.net>

aurel32 reviewed (commented):
Thanks for adding support for multiple LUT. Besides the issue reported by @vanwinkeljan and a minor nitpick, it looks all good. I have added support for the Waveshare 1.54" display, you might want to fetch the commit https://github.com/aurel32/zephyr/commit/b682c87370bcc4aa0035f022922122c88115448d in your PR.

aurel32 commented:
> Except the one small remark above I would like see the input from @galak and @MaureenHelm on this topic as there is a proposal from @galak in PR [#12457 (comment)](https://github.com/zephyrproject-rtos/zephyr/pull/12457#issuecomment-457016502) on following the linux kernel way of handling display pannels.

I think the properties of a EPD and LCD or OLED display panels are quite different. Therefore I am not sure it applies here.

jfischer-phytec-iot commented:
> > Except the one small remark above I would like see the input from @galak and @MaureenHelm on this topic as there is a proposal from @galak in PR [#12457 (comment)](https://github.com/zephyrproject-rtos/zephyr/pull/12457#issuecomment-457016502) on following the linux kernel way of handling display pannels.
> 
> I think the properties of a EPD and LCD or OLED display panels are quite different. Therefore I am not sure it applies here.

[proposal from @galak](https://github.com/zephyrproject-rtos/zephyr/pull/12457#issuecomment-457016502) could also fit EPDs. I think that what I did is not very clean :-)

vanwinkeljan commented:
Looks like in PR #12457 panel specific parameters are moved into the config struct and set at build time based on selected panel

jfischer-phytec-iot commented:
> Looks like in PR #12457 panel specific parameters are moved into the config struct and set at build time based on selected panel

How far is it better than to describe it in DT?

vanwinkeljan commented:
> How far is it better than to describe it in DT?

Not much a difference the only advantage that I can see over DT is that there is no chance in making a typo in case you need the display on another board.

Any reason to keep the rfc label as the PR looks good to me?

jfischer-phytec-iot commented:
>Any reason to keep the rfc label as the PR looks good to me?

removed RFC
